src/renate/shift/mmd_helpers.py

```diff
@@ -0,0 +1,124 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Optional, Tuple
+
+import torch
+
+
+class Kernel:
+    """Base class for kernel functions."""
+
+    def __init__(self):
+        pass
+
+    def _check_inputs(self, X0: torch.Tensor, X1: torch.Tensor):
+        assert X0.dim() == X1.dim() == 2
+        assert X0.size(1) == X1.size(1)
+        assert X0.dtype is X1.dtype
+
+
+class RBFKernel(Kernel):
```

Comment:
the kernels are not directly tied to MMD and it would be better to have them in a separate file (e.g., kernels.py)